document/src/main/java/com/ritense/document/domain/search/AdvancedSearchRequest.java

```diff
@@ -86,6 +86,19 @@ public AdvancedSearchRequest statusFilter(String status) {
         return this;
     }
 
+    public Set<String> getCaseTagsFilter() {
+        return caseTagsFilter;
+    }
+
+    public void setCaseTagsFilter(Set<String> caseTagsFilter) {
+        this.caseTagsFilter = caseTagsFilter != null ? caseTagsFilter : new HashSet<>();
+    }
+
+    public AdvancedSearchRequest caseTagsFilters(String tag) {
```

Comment:
You add 1 tag to the filter. Maybe the method name should reflect that. Something like `caseTagsFilters` or `addCaseTagsFilter`